File: Lista-15/go/7.go
package main
import "fmt"

type Professor struct {
    Nome string
}

type Escola struct {
    Nome        string
    Professores []Professor
}

func (e *Escola) AdicionarProfessor(professor Professor) {
    e.Professores = append(e.Professores, professor)
}

func main() {
    professor1 := Professor{"Carlos"}
    escola := Escola{Nome: "Escola A"}
    escola.AdicionarProfessor(professor1)

    for _, prof := range escola.Professores {
        fmt.Println(prof.Nome)
    }
}

File: Lista-15/java/7.java
import java.util.ArrayList;
import java.util.List;

class Professor {
    private String nome;

    public Professor(String nome) {
        this.nome = nome;
    }

    public String getNome() {
        return nome;
    }
}

class Escola {
    private String nome;
    private List<Professor> professores = new ArrayList<>();

    public Escola(String nome) {
        this.nome = nome;
    }

    public void adicionarProfessor(Professor professor) {
        professores.add(professor);
    }

    public List<Professor> getProfessores() {
        return professores;
    }

    public static void main(String[] args) {
        Professor professor1 = new Professor("Carlos");
        Escola escola = new Escola("Escola A");
        escola.adicionarProfessor(professor1);

        for (Professor prof : escola.getProfessores()) {
            System.out.println(prof.getNome());
        }
    }
}
